### ISSUE TITLE
Correctly highlight all the usage of the reduce tactic

### ISSUE THREAD

**Issue by favonia** [RedPRL/sml-redprl#575]

**Comment by favonia** [comment]
@cangiuli How possible it is to highlight the fancy grammar for `unfold` and `reduce`?

**Comment by cangiuli** [comment]
I can take a crack at it. The best way I can think of doing this is to have invisible regions for `unfold` and `reduce`, and to have special highlighting within those regions.

Can you explain what the possibilities are for an `elimTargetAnySort` selector?

**Comment by favonia** [comment]
Not sure what `elimTargetAnySort` mean, but a selector should be a hypothesis or `concl`, and an accessor should be `left`, `right` or `type`.
```
unfold Blah Wow Yeah at right in concl
unfold Blah Wow Yeah in x at right left
unfold Blah Wow Yeah at type
unfold Blah Wow Yeah in x concl y
unfold Blah Wow Yeah
reduce at type
reduce at right in concl
reduce in x at left
reduce in concl x y z
reduce
```

**Comment by cangiuli** [comment]
I'm referring to the following code in `redprl.grm`:
```
selector
  : CONCL (Ast.$$ (O.SEL_CONCL, []))
  | elimTargetAnySort (Ast.$$ (O.SEL_HYP, [Ast.\ ([], elimTargetAnySort)]))
```

and `unfold IntPred in eq1` in `example/omega1s1.prl`.

**Comment by favonia** [comment]
`eq1` has the same grammar as `x` in `elim x`. `elimTargetAnySort` is just some hack to make it peacefully coexist with sort-checking. Does that answer your question?

**Comment by cangiuli** [comment]
I think so. The difficulty with syntax highlighting this is figuring out where the `reduce` and `unfold` grammar ends. We could say that it ends whenever `;` or `)` or `]` is encountered. Does that sound reasonable?

**Comment by favonia** [comment]
I have to think about it. It could also be `,` as far as I can see.

**Comment by favonia** [comment]
Actually, now thinking about it, it could be `;`, `)`, `]`, `else`, `|`, `}`, `,` or maybe something weird. So, maybe just `\<else\>` or something out of [a-zA-Z0-9_]?

**Comment by cangiuli** [comment]
What do you mean "something out of `[a-zA-Z0-9_]`"?

**Comment by favonia** [comment]
`[^a-zA-Z0-9_]`

**Comment by cangiuli** [comment]
Identifiers are also allowed to contain `'` and `/` and `-`, so it gets rather complicated. But maybe this can be made to work.

Actually, it's been bothering me that `foo//` is a valid identifier: I want it to begin a comment!

**Comment by favonia** [comment]
A simpler solution is to look for `at` and highlight any `left` or `right` or `type` following it.

**Comment by cangiuli** [comment]
@favonia Has the grammar for this changed since our discussion above? I see you just made some changes to `rewrite`.

**Comment by favonia** [comment]
@cangiuli Yes, but the only change is to allow multiple accessors in `rewrite`.

**Comment by favonia** [comment]
I suppose we can always highlight words "left right type at in"?
